File: online-training-platform/leetcode/392. Is Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if(s.size() > t.size()) return false;
        string str = "";
        
        for(int i = 0, j = 0 ; i < t.size() && j < s.size() ; i++){
            if(s[j] == t[i]){
                j++;    
                str += t[i];
            }
        }
        return str == s;
    }
};

File: online-training-platform/leetcode/392. Is Subsequence.kt
class Solution {
    fun isSubsequence(s: String, t: String): Boolean {
        if(s.length > t.length){
            return false
        }

        var sIdx = 0; var tIdx = 0

        while(sIdx < s.length && tIdx < t.length){
            if(s[sIdx] == t[tIdx]) {
                sIdx += 1  
            } 
            tIdx += 1
        }

        return sIdx == s.length
    }
}